### PR TITLE
Fix ipv6 issues

### LINKED ISSUE #364: Use `NetAddress`, and remove redundant customized network types
https://github.com/kuutamolabs/lightning-knd/issues/318#issuecomment-1573434076

### LINKED ISSUE #398: IPV6 issues
IPv6 issues are from here.  Because we are using String for net address without correct type.
We are splitting the string with `:` here
https://github.com/kuutamolabs/lightning-knd/blob/1130311b6cada6c835b3d78ac3529aa9577819bf/kld/src/ldk/net_utils.rs#L35-L47

Not only the thing we pass from `kld.toml` but also other things come from the internet.
For example, if others create a fund_channel from ipv6 to us and our program will not work correctly
https://github.com/kuutamolabs/lightning-knd/blob/1130311b6cada6c835b3d78ac3529aa9577819bf/kld/src/api/channels.rs#L71-L78

Ip with a port is the socket, we should use `SocketAddr` types to handle these, and Rust standard library can help us parse it, we do not need a manual parser for a type in standard library

### PR DESCRIPTION
**Pull request by yanganto**
fix #398 
fix #364

### CODE REVIEW

File: api/Cargo.toml [hunk 1]
```diff
@@ -4,9 +4,15 @@ version = "0.1.0"
 edition = "2021"
 
 [dependencies]
+anyhow = { version = "1.0.71", features = [ "backtrace" ] }
 bitcoin = { version = "0.29.2", features = [ "serde" ] }
 serde = { version = "1.0", features = [ "derive" ] }
 serde_json = "1.0"
+lightning = { version = "0.0.115", features = ["max_level_trace", "_test_utils"] }
```
Comment (JosephGoulden) [inline api/Cargo.toml:11]: I want to try to keep the API crate simple. Perhaps we can even generate it from an OpenAPI document. So lets not add lightning dependency here.

**Review by JosephGoulden — CHANGES_REQUESTED** [review]
There is a test called `test_get_info_readonly`. Can you improve it to check that it parses the addresses properly.

File: api/Cargo.toml (hunk 1 above)
Comment (yanganto) [reply api/Cargo.toml:11]: Relate to #401

Comment (yanganto) [reply api/Cargo.toml:11]: https://github.com/kuutamolabs/lightning-knd/issues/401#issuecomment-1594293555
I already let it go, else you will see PR not issues, but the `use api::lightning::*` is the side effect from #401.

Comment (paulmattei) [reply api/Cargo.toml:11]: Is this conversation resolved? @JosephGoulden ?

Comment (yanganto) [reply api/Cargo.toml:11]: Could you open a PR to delete the `api` crate first? Such that this PR can base on that, and we do not need to wait for the swagger spec PR.

File: api/src/lib.rs [hunk 2]
```diff
@@ -102,15 +106,203 @@ pub struct GetInfo {
...
+    pub address: Vec<String>,
 }
 
-#[derive(Serialize, Deserialize, PartialEq, Debug)]
-pub struct Address {
-    #[serde(rename = "type")]
-    pub address_type: String,
-    pub address: String,
-    pub port: u16,
+/// A warper for lightning::ln::msgs::NetAddress
```
Comment (JosephGoulden) [inline api/src/lib.rs:112]: wrapper

File: api/Cargo.toml (hunk 1 above)
Comment (JosephGoulden) [reply api/Cargo.toml:11]: The swagger spec PR will generate the code in the API crate. And its probably just going to be a string, there won't be any strong typing on deserialisation. We have to do that validation in our API server. Maybe NetAddress has to live in the settings crate for now?

File: api/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (yanganto) [reply api/src/lib.rs]: Just at the same time.

File: api/Cargo.toml (hunk 1 above)
Comment (yanganto) [reply api/Cargo.toml:11]: Okay, let me do it.

**Comment by yanganto** [comment]
The test ignore also passed
```
running 1 test
test start::test_start ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 1 filtered out; finished in 45.79s
```

**Review by JosephGoulden — APPROVED** [review]

**Pull request merged**